aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/UpdateHandler.java

```diff
@@ -69,6 +80,44 @@ private void putRetentionPolicy(final AmazonWebServicesClientProxy proxy,
         logger.log(retentionPolicyMessage);
     }
 
+    private void disassociateKmsKey(final AmazonWebServicesClientProxy proxy,
+                                    final ResourceHandlerRequest<ResourceModel> request,
+                                    final Logger logger) {
+        final ResourceModel model = request.getDesiredResourceState();
+        final DisassociateKmsKeyRequest disassociateKmsKeyRequest =
+                Translator.translateToDisassociateKmsKeyRequest(model);
+        try {
+            proxy.injectCredentialsAndInvokeV2(disassociateKmsKeyRequest,
+                    ClientBuilder.getClient()::disassociateKmsKey);
+        } catch (final ResourceNotFoundException e) {
```

Comment:
[There are more exceptions that can be thrown](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_DisassociateKmsKey.html#API_DisassociateKmsKey_Errors)

Reply:
I think that it makes sense to look at the for the whole resource (other calls don't catch everything either).

Until there is a documentation page with the differences between all exceptions (I might be overlooking that - a quick google search gave me nothing), I'd probably cast the to a InternalFailure anyway

Reply:
I think it's important to throw proper exceptions because some may be retriable and some may not be. https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract-errors.html

Reply:
I agree. I'll catch them for my changes, and I'll add a follow-up issue to add the same thing to existing code